### PR TITLE
Implementing an ec-recover function for the stdlib

### PR DESCRIPTION
**Pull request by nfurfaro**
This PR adds tests for https://github.com/FuelLabs/sway-lib-std/pull/5
It also adds `sig-gen-util`, a small Rust program that starts with a hard-coded private key and a message to sign, and generates a signature, as well as the derived public key and fuel address.
There are more features to add to ec_recover which are tracked here: FuelLabs/sway-lib-std#9

### CODE REVIEW

**Comment by nfurfaro** [comment]
The CI build fails here, but will pass once this PR has been merged: https://github.com/FuelLabs/sway-lib-std/pull/5
I will add the `ec_recover` functionality in another PR once I get it working.
`ec_recover_address` works now.

File: test/src/e2e_vm_tests/test_programs/ec_recover_test/Forc.toml
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/ec_recover_test/Forc.toml:3]:
```suggestion
author = "Fuel Labs <contact@fuel.sh>"
license = "Apache-2.0"
```

File: test/src/e2e_vm_tests/test_programs/ec_recover_test/Forc.toml
Comment (adlerjohn) [inline test/src/e2e_vm_tests/test_programs/ec_recover_test/Forc.toml:9]:
```suggestion
core = { git = "http://github.com/FuelLabs/sway-lib-core" }

```

File: sig_gen_util/Cargo.toml
Comment (adlerjohn) [inline sig_gen_util/Cargo.toml:13]:
```suggestion
[workspace]

```

File: sig_gen_util/Cargo.toml
Comment (adlerjohn) [inline sig_gen_util/Cargo.toml:11]:
```suggestion
anyhow = "1.0.39"
fuel-tx = "0.1"
fuel-vm = "0.1"
hex = "0.4"
secp256k1 = { version = "0.20", features = ["recovery"] }
```

**Review by adlerjohn — COMMENTED** [review]
cargo fmt

Can you also fix clippy warnings? Try merging from master first ,they may be fixed there.

**Comment by nfurfaro** [comment]
As for clippy warnings, I can look at those but they shouldn't be related to this PR.
I'm going to merge from master 1 more time, and then address comments.

File: sig_gen_util/src/main.rs
Comment (adlerjohn) [inline sig_gen_util/src/main.rs:49]: How does this pass the formatter 😂

Comment (nfurfaro) [reply sig_gen_util/src/main.rs:49]: ahhh. sorry, there's a bunch of old commented-out code waay down there.

**Review by adlerjohn — COMMENTED** [review]
lgtm, but will defer to @sezna

File: sig_gen_util/Cargo.toml
Comment (sezna) [inline sig_gen_util/Cargo.toml:13]:
```suggestion
```

we don't need this here

**Review by sezna — CHANGES_REQUESTED** [review]

**Comment by sezna** [comment]
One tiny change then LGTM

File: sig_gen_util/Cargo.toml
Comment (nfurfaro) [reply sig_gen_util/Cargo.toml]: FIxed. I had to add `sig-gen-util` to the "members" subsection of the main cargo.toml.

File: sig-gen-util/Cargo.toml
Comment (adlerjohn) [inline sig-gen-util/Cargo.toml:4]: Can you add a publish = false line here? See https://github.com/FuelLabs/sway/blob/acc6c438a67c42cc913744523d2b4f3544434e13/test/Cargo.toml#L6

**Review by adlerjohn — CHANGES_REQUESTED** [review]
Can you do another merge from master?

File: sig-gen-util/Cargo.toml
Comment (nfurfaro) [reply sig-gen-util/Cargo.toml:4]: Done.

**Comment by adlerjohn** [comment]
You'll have to do another merge from master to pick up https://github.com/FuelLabs/sway/pull/618

**Review by adlerjohn — APPROVED** [review]

**Review by sezna — APPROVED** [review]

**Pull request merged**
